tests/unit/test_gce_authority.py

```diff
@@ -0,0 +1,188 @@
+    expected_metadata_blacklist = [
+        dict([pair]) for pair in authority_config['metadata_blacklist']
+    ]
+    gce_client.assert_called_once_with(
+        auth_client(),
+        session,
+        tag_blacklist=authority_config['tag_blacklist'],
+        metadata_blacklist=expected_metadata_blacklist)
+
+    assert crm_client() is authority.crm_client
+    assert gce_client() is authority.gce_client
+    assert rrset_channel is authority.rrset_channel
+    assert authority_config is authority.config
+
+
+@pytest.mark.asyncio
+async def test_builder_error_on_required_config(authority_config, caplog):
+    caplog.set_level(logging.ERROR)
+    del authority_config['crm_keyfile']
+    rrset_channel = 'channel'
+    kwargs = {}
+
+    builder = gce_authority.GCEAuthorityBuilder(
+        authority_config, rrset_channel, **kwargs)
+    session = builder.session
+    with pytest.raises(exceptions.GCPConfigError), contextlib.closing(session):
+        builder.build_authority()
+    assert 1 == len(caplog.records)
+
+
+@pytest.mark.asyncio
+async def test_start_publishes_msg_to_channel(mocker, authority_config,
+                                              get_gce_client, create_mock_coro):
+    instance_data = [{
+        'hostname': f'host-{i}',
+        'internal_ip': f'192.168.1.{i}',
+        'external_ip': f'1.1.1.{i}'
+    } for i in range(1, 4)]
+    active_projects_mock, active_projects_coro = create_mock_coro()
+    active_projects_mock.return_value = [{'projectId': 1}, {'projectId': 2}]
+    crm_client = get_gce_client(gcp_clients.GCRMClient)
+    crm_client.list_all_active_projects = active_projects_coro
+
+    list_instances_mock, list_instances_coro = create_mock_coro()
+    list_instances_mock.return_value = instance_data
+    gce_client = get_gce_client(gcp_clients.GCEClient)
+    gce_client.list_instances = list_instances_coro
+
+    msg_gen = msg_generators.ARecordMsgGenerator(authority_config['zone'])
+    rrset_channel = asyncio.Queue()
+    authority = gce_authority.GCEAuthority(
+        authority_config, crm_client, gce_client, msg_gen, rrset_channel)
+
+    await authority.start()
+
+    expected_rrsets = [{
+        'name': record['hostname'],
+        'type': 'A',
+        'rrdatas': [record['internal_ip']]
+    } for record in instance_data]
+    expected_msg = {
+        'zone': authority_config['zone'],
+        'rrsets': expected_rrsets
+    }
+
+    # one message per each project
+    assert (await rrset_channel.get()) == expected_msg
+    assert (await rrset_channel.get()) == expected_msg
+    # start also calls self.done at the end
+    assert (await rrset_channel.get()) is None
+
+
+def test_get_gce_authority_entrypoint(mocker, authority_config):
+    target = 'gordon_janitor_gcp.gce_authority.GCEAuthorityBuilder'
+    authority_obj = mocker.Mock()
+    authority_class = mocker.MagicMock(
+        return_value=authority_obj)
+    mocker.patch(target, new=authority_class)
```

Comment:
Actually, I'm not sure I understand the whole need to patch this if you're calling `get_authority.get_authority` directly?